### DISCUSSION TITLE
Is there a way to get the entire POST body without having to assemble it oneself?

### DISCUSSION THREAD

**Discussion by clansdown** [uNetworking/uWebSockets#1533] [Q&A]
Since there are a lot of use cases where you can't do anything with a POST request until you've got the entire body in (e.g. a json request) is there a lambda that one can hang onto the request for handling only the entire post body, or does one just have to have the code to assemble the body over multiple onData invocations? If so, since assembling the POST body prior to doing anything is going to be identical every time, is there a standard helper function to provide the lamba to do the buffer assembly?

**Comment by uNetworkingAB** [comment]
No such thing built in

**Reply by clansdown** [reply]
Can you give any guidance on how to assemble the body into a single buffer without a memory leak if the POST request has chunks but then gets aborted? Is there any way to hang data and a deleter function onto a request to guarantee no memory leaks at the end of the request lifecycle, regardless of how it terminates?

**Reply by uNetworkingAB** [reply]
Oh there is no memory leak. You capture a std::string and just append to it. Raii will manage everything for you

**Reply by clansdown** [reply]
That's not what the example in Http3Server does. Is it better to do something like
std::string buffer;
res->onData([res, buffer], etc
to capture the std::string by value, rather than the lazy-initialization in the Http3Server example?

**Reply by uNetworkingAB** [reply]
It's not for me to decide how you append strings together. Like said, you can capture a std::string and append to it. RAII will free your memory. I really cannot micro manage what people decide to do here, consider it artistic freedom. It's not rocket science either way; you can do it in no more than 3 lines of abstract code if you want

**Reply by clansdown** [reply]
I'm sorry for not writing clearly, I meant to make sure I understood what you were saying, not to ask you to make decisions for me. Another way of asking what I meant to ask is whether I'm wrong about the Http3Server example in the distribution having a potential memory leak on the onAbort code not freeing the buffer if chunks have already come in, or if I am making a mistake in my analysis because I have missed something.
